### PR TITLE
refactor: improve ZETA deposit check with max supply check

### PR DESCRIPTION
**Pull request by lumtis**
# Description

Add a check for ZETA deposit to not surpass a supply cap as an extra security check

## Release Notes

- **New Features**
  - Introduced a method to retrieve the supply of specific tokens, enhancing the functionality of the banking interface.
  - Added a new error message for when the maximum supply of tokens is reached.

- **Bug Fixes**
  - Improved error handling in minting processes to prevent exceeding the defined maximum supply of tokens.

- **Tests**
  - Expanded test coverage for minting functionality, including edge cases for maximum supply limits and error handling scenarios.

### CODE REVIEW

**Review by fbac — APPROVED** [review]

File: x/fungible/keeper/zeta.go [hunk 1]
```diff
@@ -9,9 +10,24 @@
...
+	zetaMaxSupply, ok := sdk.NewIntFromString(ZETAMaxSupplyStr)
+	if !ok {
+		return errors.New("failed to parse ZETA max supply")
+	}
+
+	// Check if the max supply is reached
+	supply := k.bankKeeper.GetSupply(ctx, config.BaseDenom)
+	if supply.Amount.Add(sdk.NewIntFromBigInt(amount)).GT(zetaMaxSupply) {
+		return types.ErrMaxSupplyReached
+	}
```
[the remark this answers is not in the document]
Comment (swift1337) [reply x/fungible/keeper/zeta.go:29]: agree

File: x/fungible/keeper/zeta.go [hunk 2]
```diff
@@ -23,6 +39,17 @@ func (k *Keeper) MintZetaToEVMAccount(ctx sdk.Context, to sdk.AccAddress, amount
 }
 
 func (k *Keeper) MintZetaToFungibleModule(ctx sdk.Context, amount *big.Int) error {
+	zetaMaxSupply, ok := sdk.NewIntFromString(ZETAMaxSupplyStr)
+	if !ok {
+		return errors.New("failed to parse ZETA max supply")
+	}
+
+	// Check if the max supply is reached
+	supply := k.bankKeeper.GetSupply(ctx, config.BaseDenom)
```
Comment (swift1337) [inline x/fungible/keeper/zeta.go:48]: same here

**Review by swift1337 — APPROVED** [review]
Left a comment, other than that all good

File: x/fungible/keeper/zeta.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (lumtis) [reply x/fungible/keeper/zeta.go:29]: I will apply for develop here for code cleanliness but I think it's fine the parsing error is more a developer error handling since we handle a constant
